extensions/tags/src/Query/TagFilterGambit.php

```diff
@@ -48,22 +48,27 @@ public function getFilterKey(): string
 
     public function filter(FilterState $filterState, string $filterValue, bool $negate)
     {
-        $this->constrain($filterState->getQuery(), $filterValue, $negate);
+        $this->constrain($filterState->getQuery(), $filterValue, $negate, $filterState->getActor());
     }
 
-    protected function constrain(Builder $query, $rawSlugs, $negate)
+    protected function constrain(Builder $query, $rawSlugs, $negate, User $actor)
     {
         $slugs = explode(',', trim($rawSlugs, '"'));
 
-        $query->where(function (Builder $query) use ($slugs, $negate) {
+        $query->where(function (Builder $query) use ($slugs, $negate, $actor) {
             foreach ($slugs as $slug) {
                 if ($slug === 'untagged') {
                     $query->whereIn('discussions.id', function (Builder $query) {
                         $query->select('discussion_id')
                             ->from('discussion_tag');
                     }, 'or', ! $negate);
                 } else {
-                    $id = $this->tags->getIdForSlug($slug);
+                    // @TODO: grab all IDs first instead of multiple queries.
```

Comment:
We might want to eventually add a search method to our slug driver interface, which would support one or more "OR"d arguments, with partial matching on each.

Reply:
yup! we would probably have to add it in a major release as it'd be changing the interface.

(this review caught me off guard 😱)

Reply:
The reviews will start trickling back in 🤗